src/time/Waveform.cpp

```diff
@@ -88,11 +98,11 @@ Eigen::VectorXd Waveform::extrapolateData(int order, int timeWindows)
 Eigen::VectorXd Waveform::interpolateData(int order, int timeWindows, double dt)
 {
   Eigen::VectorXd interpolatedValue;
-  if (order == 0) {
+  if ((order == 0) || (timeWindows < 2 && order > 0)) {
     // constant interpolation = just use sample at the end of the window: x(dt) = x^t
     PRECICE_ASSERT(this->numberOfSamples() > 0);
     interpolatedValue = this->_timeWindows.col(0);
-  } else if ((order == 1) || (timeWindows < 2 && order > 1)) {
+  } else if ((order == 1) || (timeWindows < 3 && order > 1)) {
```

Comment:
Similar to the change above, but for relevant for second order interpolation and the second window.